src/Graph.ts

```diff
@@ -825,6 +835,8 @@ Try defining "${chunkName}" first in the manualChunks definitions of the Rollup
 	}
 
 	warn(warning: RollupWarning) {
+		if (this.shouldFilterWarning && this.shouldFilterWarning(warning)) return;
```

Comment:
I was thinking it might be worthwhile to move this into the makeOnWarn function so that custom warning handlers still get all the warnings.

Perhaps the knowledge of filters could be shared through a second argument to warn - `filtered: true / false`.

Reply:
Interesting! I like the idea of passing the filter through.

Reply:
Speaking of `makeOnWarn` I feel like it's relationship to this code is very strange to me
https://github.com/rollup/rollup/blob/master/src/utils/mergeOptions.ts#L40-L46

Shouldnt that be where we use makeOnWarn? and that would give a clearer spot to create the final warning handler?

Reply:
`getOnWarn()` also uses the `silent` option, which seems sort of overlapping what we are doing now. Is that strange? Maybe what we should be doing is extending silent to take warning codes as arguments?

Reply:
Interesting, I guess this is due to bootstrapping that there is a phase before plugins have been loaded in options where we still need an initial warning handler - but yes I don't see why that can't just be the same handler as makeOnWarn. Unifying these two seems sensible to me.

Reply:
Perhaps just pass `isSilenced` as a second argument to avoid warning mutation?

Reply:
--silent as a boolean flag but --skipWarning as taking an argument seems worth keeping them separate for? But a single `silenced` argument could still be passed in the warning handler for both cases?

Reply:
My hunch is that the reason `makeOnWarn` is used inside the Graph constructor is because it stores a list of duplicate warnings. So maybe that needs to be recreated at certain points to prevent removing valuable warnings. Maybe in watch mode or something?

Reply:
And if that is the case I think I can deal with that in a different way

Reply:
The custom warning handler currently takes the default handler as a second arg here: https://github.com/rollup/rollup/blob/master/src/utils/mergeOptions.ts#L56

I suppose it could be a third argument